colorbleed/plugins/maya/publish/validate_animation_out_set_related_node_ids.py

```diff
@@ -3,87 +3,127 @@
 import pyblish.api
 import colorbleed.api
 import colorbleed.maya.lib as lib
-import avalon.io as io
 
 
-class ValidateAnimationOutSetRelatedNodeIds(pyblish.api.InstancePlugin):
-    """Validate rig out_SET nodes have related ids to current context
+def get_parent(node):
+    """Get the parent node of the given node
+    Args:
+        node (str): full path of the node
 
-    An ID is 'related' if its built in the current Item.
+    Returns:
+        str, full path if parent node
+    """
+    return cmds.listRelatives(node, parent=True, fullPath=True)
+
+
+class ValidateOutRelatedNodeIds(pyblish.api.InstancePlugin):
+    """Validate if deformed shapes have related IDs to the original shapes
 
-    Note that this doesn't ensure it's from the current Task. An ID created
-    from `lookdev` has the same relation to the Item as one coming from others,
-    like `rigging` or `modeling`.
+    Any intermediate shapes which are created when creating deformers on
```

Comment:
```
When a deformer is applied in the scene on a referenced mesh that already had
deformers then Maya will create a new shape node for the mesh that does not
have the original id. This validator checks whether the ids are valid on all the shape
nodes in the instance.
```

Does that make more sense as docstring?

Reply:
It explains it a bit more, I will but it as docstrings